pkg/ccl/importccl/import_stmt.go

```diff
@@ -619,20 +623,54 @@ func importPlanHook(
 
 		telemetry.CountBucketed("import.files", int64(len(files)))
 
-		_, errCh, err := p.ExecCfg().JobRegistry.CreateAndStartJob(ctx, resultsCh, jobs.Record{
+		// Here we create the job and protected timestamp records in a side
+		// transaction and then kick off the job. This is awful. Rather we should be
+		// disallowing this statement in an explicit transaction and then we should
+		// create the job in the user's transaction here and then in a post-commit
+		// hook we should kick of the StartableJob which we attached to the
+		// connExecutor somehow.
+		protectedTimestampID := uuid.MakeV4()
+		jr := jobs.Record{
 			Description: jobDesc,
 			Username:    p.User(),
 			Details: jobspb.ImportDetails{
-				URIs:       files,
-				Format:     format,
-				ParentID:   parentID,
-				Tables:     tableDetails,
-				SSTSize:    sstSize,
-				Oversample: oversample,
-				SkipFKs:    skipFKs,
+				URIs:                     files,
+				Format:                   format,
+				ParentID:                 parentID,
+				Tables:                   tableDetails,
+				SSTSize:                  sstSize,
+				Oversample:               oversample,
+				SkipFKs:                  skipFKs,
+				ProtectedTimestampRecord: protectedTimestampID,
 			},
 			Progress: jobspb.ImportProgress{},
-		})
+		}
+		var sj *jobs.StartableJob
+		// Prepare the protected timestamp record.
+		spans := make([]roachpb.Span, 0, len(tableDetails))
+		for i := range tableDetails {
+			spans = append(spans, tableDetails[i].Desc.TableSpan())
```

Comment:
I think we only need to add spans for tables where `IsNew` is false (which could require revert) and then below we can skip if spans is empty